bitcoind-rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindInstance.scala

```diff
@@ -166,27 +195,68 @@ object BitcoindInstance extends InstanceFactory[BitcoindInstance] {
     fromConfig(conf, binary)
   }
 
-  override def fromConfigFile(file: File): BitcoindInstance = {
-    fromConfFile(file, DEFAULT_BITCOIND_LOCATION)
+  override def fromConfigFile(file: File): BitcoindInstanceLocal = {
+    fromConfFile(
+      file,
+      DEFAULT_BITCOIND_LOCATION match {
+        case Some(file) => file
+        case None       => bitcoindLocationFromConfigFile
+      }
+    )
   }
 
   /** Constructs a `bitcoind` instance from the given config */
   def fromConfig(
       config: BitcoindConfig,
-      binary: File = DEFAULT_BITCOIND_LOCATION
-  ): BitcoindInstance = {
+      binary: File = DEFAULT_BITCOIND_LOCATION match {
+        case Some(file) => file
+        case None       => bitcoindLocationFromConfigFile
+      }
+  ): BitcoindInstanceLocal = {
 
     val authCredentials = BitcoindAuthCredentials.fromConfig(config)
-    BitcoindInstance(config.network,
-                     config.uri,
-                     config.rpcUri,
-                     authCredentials,
-                     zmqConfig = ZmqConfig.fromConfig(config),
-                     binary = binary,
-                     datadir = config.datadir)
+    BitcoindInstanceLocalImpl(config.network,
+                              config.uri,
+                              config.rpcUri,
+                              authCredentials,
+                              zmqConfig = ZmqConfig.fromConfig(config),
+                              binary = binary,
+                              datadir = config.datadir)
   }
 
   override val DEFAULT_DATADIR: Path = BitcoindConfig.DEFAULT_DATADIR.toPath
 
   override val DEFAULT_CONF_FILE: Path = BitcoindConfig.DEFAULT_CONF_FILE.toPath
 }
+
+object BitcoindInstanceRemote extends InstanceFactory[BitcoindInstanceRemote] {
+
+  private case class BitcoindInstanceRemoteImpl(
+      network: NetworkParameters,
+      uri: URI,
+      rpcUri: URI,
+      authCredentials: BitcoindAuthCredentials,
+      zmqConfig: ZmqConfig,
+      proxyParams: Option[Socks5ProxyParams]
+  ) extends BitcoindInstanceRemote
+
+  def apply(
+      network: NetworkParameters,
+      uri: URI,
+      rpcUri: URI,
+      authCredentials: BitcoindAuthCredentials,
+      zmqConfig: ZmqConfig = ZmqConfig(),
+      proxyParams: Option[Socks5ProxyParams] = None
+  ): BitcoindInstanceRemote = {
+    BitcoindInstanceRemoteImpl(network,
+                               uri,
+                               rpcUri,
+                               authCredentials,
+                               zmqConfig = zmqConfig,
+                               proxyParams = proxyParams)
+  }
+
+  override def fromConfigFile(file: File): BitcoindInstanceRemote = ???
```

Comment:
This needs to be implemented, you can copy `BitcoindInstanceLocal.fromConfigFile()` i believe 

https://github.com/shruthii625/bitcoin-s/blob/56dd191d55452c16bcbc4a473b448e42e1d50775/bitcoind-rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindInstance.scala#L198